src/components/Question.jsx

```diff
@@ -107,28 +120,42 @@ export default class Question extends React.Component {
 
   render() {
     const question = this.props.question;
+    const subQuestion = question[Constants.HAS_SUBQUESTION];
     const options = this.context.options;
-
+    const renderQuestion = this.questionComponent(question);
```

Comment:
noooooooooo :) 

opposite way around :). You have two things in this code:
1) object called questionComponent
2) function called renderQuestion with parameter question

And you just call the function to retrieve the object:
```
const questionComponent = this.renderQuestion(question);
```

I.e.:
-  functions always do: "render question", "get question", "save question"
- objects always are something: "question component", "rendered question", "question", "saved question"